### PR TITLE
[RNTuple] write multiple fields 

### PR DESCRIPTION
**Pull request by Moelf**
```julia
Nitems = 10
       data = [5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
       newtable = Dict(
               "x1" => Float64.(data),
               "x2" => Float32.(data),
               "x3" => Int32.(data),
               "y1" => UInt16.(data),
           )
       UnROOT.write_rntuple(open("/tmp/a.root", "w"), newtable; rntuple_name="myntuple")
       LazyTree("/tmp/a.root", "myntuple")
 Row │ x1       y1      x2       x3
     │ Float64  UInt16  Float32  Int32
─────┼─────────────────────────────────
 1   │ 5.0      5       5.0      5
 2   │ 6.0      6       6.0      6
 3   │ 7.0      7       7.0      7
 4   │ 8.0      8       8.0      8
 5   │ 9.0      9       9.0      9
 6   │ 10.0     10      10.0     10
 7   │ 11.0     11      11.0     11
 8   │ 12.0     12      12.0     12
 9   │ 13.0     13      13.0     13
 10  │ 14.0     14      14.0     14
```

## ROOT is now happy 
```
$ root --version
ROOT Version: 6.33.01
Built for linuxx8664gcc on Oct 03 2024, 22:53:26
From heads/master@v6-31-01-3492-g65de35a80f
```
```python
$ python
Python 3.11.9 (main, Jun 24 2024, 14:32:54) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ROOT
df  =  ROOT.RDataFrame("myntuple", "/tmp/a.root")
[ROOT.NTuple] Warning /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/tree/ntuple/v7/src/RPageStorageFile.cxx:326 in ROOT::Experimental::Internal::RPageSourceFile::LoadStructureImpl()::<lambda()>:0: RuntimeWarning: Pre-release format version: RC 2
>>> df.GetColumnNames()
vector<string>{ "x1", "x2", "x3", "y1" }
>>> list(df.Take['std::int32_t']("x3"))
[5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
>>> list(df.Take['float']("x2"))
[5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0]
```
- [x] Do round trips within UnROOT.jl
- [x] make ROOT happy

### CODE REVIEW

**Comment by Moelf** [comment]
the current "C++ ROOT read back" CI setup is a bit janky, it will be better once there exists a ROOT release with 1.0.0 RNTuple spec.

I also would want to move to using Julia's Artifacts system for distribution test `.root` files, but for now maybe this is good enough.

Another thing is I bumped version of Julia and StaticArrays, @tamasgal can you let me know if you guys all moved to at least 1.10?

**Comment by tamasgal** [comment]
Yes that's fine, keep moving!

File: src/RNTuple/Writing/TFileWriter.jl [hunk 1]
```diff
@@ -505,30 +531,33 @@ function write_rntuple(file::IO, table; file_name="test_ntuple_minimal.root", rn
...
-    rntAnchor_update[:fSeekHeader] = UInt32(position(file))
-    rnt_header = UnROOT.RNTupleHeader(zero(UInt64), rntuple_name, "", "ROOT v6.33.01", [
-    UnROOT.FieldRecord(zero(UInt32), zero(UInt32), zero(UInt32), zero(UInt16), zero(UInt16), 0, -1, -1, string(only(input_schema.names)), RNTUPLE_WRITE_TYPE_CPPNAME_DICT[input_T], "", ""),
-    ], [UnROOT.ColumnRecord(RNTUPLE_WRITE_TYPE_IDX_DICT[input_T]..., zero(UInt32), 0x00, 0x00, 0),], UnROOT.AliasRecord[], UnROOT.ExtraTypeInfo[])
+    RBlob1 = UnROOT.RBlob(; fNbytes = 0x00DC, fVersion = 0x0004, fObjLen = 0x000000BA, fDatime = Stubs.WRITE_TIME, fKeyLen = 34,
+    fCycle = 0x0001, fSeekKey = position(file), fSeekPdir = 100, fClassName = "RBlob", fName = "", fTitle = "")
+    RBlob1_update = Dict{Symbol, Any}()
+    RBlob1_obs = rnt_write_observe(file, RBlob1)
+    field_records, col_records = schema_to_field_column_records(table)
+    rnt_header = UnROOT.RNTupleHeader(
```
Comment (tamasgal) [inline src/RNTuple/Writing/TFileWriter.jl:539]: I got a bit lost with the dense code lines here, maybe put a little bit of white space and indents 😅

File: src/RNTuple/Writing/page_writing.jl [hunk 2]
```diff
@@ -1,48 +1,82 @@
...
-function rnt_ary_to_page(ary::AbstractVector{Int16})
+function rnt_ary_to_page(ary::AbstractVector{Int16}, cr::ColumnRecord)
+    (;split, zigzag, delta) = _detect_encoding(cr.type)
     Page_write(reinterpret(UInt8, ary))
 end
 
-function rnt_ary_to_page(ary::AbstractVector{Int8})
+function rnt_ary_to_page(ary::AbstractVector{Int8}, cr::ColumnRecord)
+    (;split, zigzag, delta) = _detect_encoding(cr.type)
     Page_write(reinterpret(UInt8, ary))
```
Comment (tamasgal) [inline src/RNTuple/Writing/page_writing.jl:80]: Maybe add a ˋPage_writeˋ  method which does a catch all and reinterpret to ˋUint8ˋ for everything else.

**Review by tamasgal — APPROVED** [review]

File: src/RNTuple/Writing/TFileWriter.jl (hunk 1 above)
Comment (Moelf) [reply src/RNTuple/Writing/TFileWriter.jl:539]: yea this part is totally manual right now.

we build each block of things as they appear in the file, and manually construct "observable"-like structure for each of them.

eventually they should be automated once we're more certain what kind of flexibility is needed

**Pull request merged**